### ISSUE TITLE
Terraform Google Provider proposes Big Query Table replace

### ISSUE THREAD

**Issue by anand-monster** [hashicorp/terraform-provider-google#10130]
A column is dropped from Terraform Script and when terraform plan is run it proposes to replace the table instead of just and update/alter to existing table 

module.<project big query table>.google_bigquery_table.bq_table must be replaced

Following are the version details for the provider.

Initializing provider plugins...
- Finding latest version of hashicorp/google...
- Finding latest version of hashicorp/google-beta...
- Installing hashicorp/google v3.85.0...
- Installed hashicorp/google v3.85.0 (signed by HashiCorp)
- Installing hashicorp/google-beta v3.85.0...
- Installed hashicorp/google-beta v3.85.0 (signed by HashiCorp)

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Affected Resource(s)

* google_XXXXX

### Terraform Configuration Files

```tf
# Copy-paste your Terraform configurations here.
#
# For large Terraform configs, please use a service like Dropbox and share a link to the ZIP file.
# For security, you can also encrypt the files using our GPG public key:
#    https://www.hashicorp.com/security
#
# If reproducing the bug involves modifying the config file (e.g., apply a config,
# change a value, apply the config again, see the bug), then please include both:
# * the version of the config before the change, and
# * the version of the config after the change.
```

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by nbali** [comment]
fyi https://github.com/hashicorp/terraform-provider-google/issues/12441

**Comment by obada-ab** [comment]
@melinath this one's a bit old but it should be a feature request not a bug, and it would be fixed by https://github.com/GoogleCloudPlatform/magic-modules/pull/10170

**Comment by melinath** [comment]
Agreed. Duplicate of https://github.com/hashicorp/terraform-provider-google/issues/14769
